### ISSUE TITLE
Should not issue error on property assignments to function/class declarations in block scopes.

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#16271]
Reported by @OliverJAsh in #16239:

```ts
{
class Foo {}
Foo.bar = 1; // error

const Foo2 = () => {};
Foo2.bar = 1; // error

function FooBar() {}
FooBar.bar = 1; // error
}
```

None of these should error in JS checking.  This may have to do with how we resolve function declarations in block bodies (i.e. we look them up in the containing function scope, not the block scope.

**Comment by OliverJAsh** [comment]
Thanks @DanielRosenwasser. Note the `Foo2` example (function expression) also errors in the root scope.

**Comment by jakebailey** [comment]
Just looking at ancient issues; this one was fixed by #38031 (and I believe is tested there by nature of the tests having no errors).
